src/RabbitMQ/AmqpTemplate.php

```diff
@@ -18,16 +22,63 @@
     protected $channel;
 
     /**
+     * @var string
+     */
+    protected $escapeMode;
+
+    /**
      * Constructor
      *
      * @param string $host
      * @param int    $port
      * @param string $user
      * @param string $pass
+     * @param string $escapeMode
      */
-    public function __construct($host, $port, $user, $pass)
+    public function __construct($host, $port, $user, $pass, $escapeMode = self::ESCAPE_MODE_SERIALIZE)
     {
         $this->connection = new AMQPConnection($host, $port, $user, $pass);
         $this->channel = $this->connection->channel();
+        $this->escapeMode = $escapeMode;
+    }
+
+    /**
+     * Escape the message
+     *
+     * @param $message
+     * @return string
+     */
+    protected function escape($message)
```

Comment:
Yes, could have done that but as you said, it would have been more complex for no (or little gain)... maybe for the v3...